Dockerfile

```diff
@@ -1,94 +1,115 @@
 FROM ubuntu:14.04
 MAINTAINER support@civisanalytics.com
 
-# Ensure UTF-8 locale.
-RUN locale-gen en_US.UTF-8
+# We need this file to install the right version of numpy.
+COPY requirements.txt /requirements.txt
```

Comment:
This is the part of the image which will change most frequently. Let's put it near the end to avoid recompiling Python with every package version change. I think you can combine the 
```
pip3 install --no-cache-dir `grep 'numpy' /requirements.txt` && \
 +    find /usr/local -depth \
 +        \( \
 +            \( -type d -a -name test -o -name tests \) \
 +            -o \
 +            \( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
 +        \) -exec rm -rf '{}' + && \
 +    rm -rf /usr/src/python ~/.cache
```
onto the same layer with the other pip installs, rather than in the Python installation layer.